### PR TITLE
fix: set current dir in nvim

### LINKED ISSUE #2077: bug: extension fails to load when using neovim under wsl
### Check the following:

- [X] I have checked the behavior after setting "vscode-neovim.neovimClean" in VSCode settings.
- [X] I have read the vscode-neovim docs.
- [X] I have searched existing issues.

### Neovim version (nvim -v)

v0.9.1

### Operating system/version

Windows 11 Enterprise 23H2 22631.3593

### Describe the bug

Everything has been working beautifully until I installed the latest update (1.15.1)
Now the extension fails to load with the following error message:

> [Failed to start nvim] Neovim spawn error: spawn
> C:\Windows\system32\wsl.exe ENOENT
> source: VSCode Neovim

Extension Host logs this error:
``` 
2024-06-05 17:28:47.149 [error] Error: [ExtensionListenerError] Extension 'asvetliakov.vscode-neovim' FAILED to handle event: TypeError: Cannot read properties of undefined (reading 'command')
	at n.value (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:163:20749)
	at l.y (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:664)
	at l.fire (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:881)
	at C.onDidChangeWindowProperty (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:145:21077)
	at C.$onDidChangeWindowFocus (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:145:20888)
	at E.S (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:5970)
	at E.Q (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:5736)
	at E.M (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:4762)
	at E.L (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:3589)
	at n.value (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:150:2282)
	at l.y (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:664)
	at l.fire (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:881)
	at s.fire (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:107:14502)
	at n.value (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:176:8481)
	at l.y (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:664)
	at l.fire (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:82:881)
	at s.fire (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:107:14502)
	at MessagePortMain.<anonymous> (c:\Users\amoore\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:176:6631)
	at MessagePortMain.emit (node:events:514:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:2285)
```

**I rolled back to the previous version (1.15.0) and it works just fine.**

### Steps To Reproduce

1. Install latest version of plugin
2. Configure it to use WSL with default settings. I only have one distribution installed.
3. Configure Neovim Executable Paths: linux. For my setup nvim is on the path so I just put 'nvim'

### Expected Behavior

Plug loads successfully.

### PR DESCRIPTION
**Pull request by xiyaowong**
fix #2077

nvim may in the different fs

### CODE REVIEW

File: src/main_controller.ts [hunk 1]
```diff
@@ -381,9 +362,36 @@ export class MainController implements vscode.Disposable {
...
+                cwd = expectedCwd;
+            }
+        } else {
+            cwd = config.useWsl ? wslpath(expectedCwd) : expectedCwd;
+        }
+
+        if (cwd) {
+            logger.debug(`Setting current dir to: ${cwd}`);
+            try {
+                await this.client.request("nvim_set_current_dir", [cwd]);
```
Comment (ollien) [inline src/main_controller.ts:388]: I don't have WSL setup to test with, but why is this different from setting the CWD of the spawned process?

Comment (xiyaowong) [reply src/main_controller.ts:388]: When using WSL, we are actually spawning wsl.exe

https://github.com/vscode-neovim/vscode-neovim/blob/4d1ccc93e53631547d4cb89dbf1f29e579775621/src/main_controller.ts#L200-L205

**Review by ollien — APPROVED** [review]

File: src/main_controller.ts [hunk 2]
```diff
@@ -381,9 +362,36 @@ export class MainController implements vscode.Disposable {
...
+        if (vscode.env.remoteName) {
+            // Runs on the Remote Extension Host
+            if (this.extContext.extension.extensionKind === ExtensionKind.Workspace) {
+                cwd = expectedCwd;
+            }
+        } else {
+            cwd = config.useWsl ? wslpath(expectedCwd) : expectedCwd;
+        }
+
+        if (cwd) {
```
Comment (justinmk) [inline src/main_controller.ts:385]: if cwd is falsey, should log a warning at least

File: src/main_controller.ts [hunk 3]
```diff
@@ -381,9 +362,36 @@ export class MainController implements vscode.Disposable {
     private setClientInfo() {
         const versionString = this.extContext.extension.packageJSON.version as string;
         const [major, minor, patch] = [...versionString.split(".").map((n) => +n), 0, 0, 0];
+        logger.debug(`Setting client info: vscode-neovim ${major}.${minor}.${patch}`);
         this.client.setClientInfo("vscode-neovim", { major, minor, patch }, "embedder", {}, {});
     }
 
+    private async setCurrentDir() {
```
Comment (justinmk) [inline src/main_controller.ts:369]: neither the linked issue, nor this PR , nor the commit message, explain why this works but setting `cwd` in the spawn() call doesn't. A docstring on this function would be helpful.

In the future, please explain the problem that is being solved by each PR, in the commit message.

Comment (xiyaowong) [reply src/main_controller.ts:369]: image

**Pull request merged**